### DISCUSSION TITLE
ERROR: unable to load demographics

### DISCUSSION THREAD

**Discussion by nourabdelbaki** [synthetichealth/synthea#1464] [Q&A] [answered]
Hi everyone,

I am trying to change **as little as possible** to run a simulation for my country's demographics. So, I followed instructions and changed the demographics file, zipcodes, providers, and timezones files. I thought accuracy does not matter, so I just made sure there is an instance of my city and state in all of the files under providers, zipcodes, and timezones. 

How my demographics file looks like: (after header)
"1,County1,Region1,Egypt,10000,Region1,10000,0.505672041767544,0.494327967242268,0.27,0.095,0.27,0.095,0.27,0.0,0.136294028,0.111026034,0.09532925,0.095137825,0.086906585,0.086140888,0.077431087,0.068338438,0.05426876,0.046228943,0.042113323,0.034360643,0.027852221,0.018089587,0.010719755,0.009762634,0.0,0.0,0.4,0.251,0.25,0.25,0.105,0.105,0.016,0.016,0.02,0.02,0.136906586,0.671564713,0.129849701,0.061679001"

Zipcodes:
',Egypt,EG,Region1,,33.091627,-85.527029'
Example from providers, hospitals:
""112011","1588664007","SELECT SPECIALTY HOSPITAL - SAVANNAH, INC","5353 REYNOLDS STREET 4 SOUTH","Region1",EG,"314056015","051",32.0564572,-81.0951271,"9128197986","00-09","01-02",false,,,"04",40,"

I just made similar changes to all's first row to have Region1, EG for city, state. 

When running:
`./run_synthea -s 123 -p 100 Egypt Region1`

I get the following error:
`> Task :run FAILED
ERROR: unable to load demographics
java.lang.Exception: The city Region1 was not found in the demographics file for state Egypt.
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:74)
        at org.mitre.synthea.engine.Generator.init(Generator.java:228)
        at org.mitre.synthea.engine.Generator.<init>(Generator.java:209)
        at App.main(App.java:270)
Exception in thread "main" java.lang.ExceptionInInitializerError
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:102)
        at org.mitre.synthea.engine.Generator.init(Generator.java:228)
        at org.mitre.synthea.engine.Generator.<init>(Generator.java:209)
        at App.main(App.java:270)
Caused by: java.lang.Exception: The city Region1 was not found in the demographics file for state Egypt.
        at org.mitre.synthea.world.geography.Location.<init>(Location.java:74)
        ... 3 more

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk-11.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s
4 actionable tasks: 3 executed, 1 up-to-date`

I have also made the appropriate changes in the synthea.properties file under main > resources

I am not sure what is going wrong? Again, I just want to change the demographics, but I do not care about anything else. Is it an accuracy problem? Do all of the locations have to align? If so, can anyone let me know what is the easiest way to just change demographics? Can I just add arbitrary providers that cover the entire city/state? 

I hope this makes sense!

Thanks a lot in advance!

**Comment by shabiel** [comment]
Check https://github.com/synthetichealth/synthea-international.

I am curious what use is Synthea being used for in Egypt. I was involved in consultations for EMRs in Egypt a while back.

**Reply by nourabdelbaki** [reply]
Hi Sam, 
It is not for Egypt per se, it is for a project I am doing for a class in my master's.

**Comment by jawalonoski** [comment]
@nourabdelbaki Not sure what version of Synthea you're using, but I get a slightly different error with your configuration changes when using the latest code on `master`.

At any rate... You're using "Egypt" as a state, although it is a country, so in some of the output it shows the country as "US". Nevertheless, if you want to make your changes work, you need to change a few more files.

- You already got these...
  - geography/demographics.csv
  - geography/zipcodes.csv
  - providers/hospitals.csv
- Additional requirements...
  - geography/sdoh.csv
  - payers/eligibility_input_files/medicaid_income_eligibilities.csv
  - payers/eligibility_input_files/medicaid_mnil_eligibilities.csv

You might also see occasional errors with other files... for example,
  - providers/snf.csv

**Reply by nourabdelbaki** [reply]
Thank you for your response! 

For the additional requirements and especially, medicaid eligibility files, if I want to do a universal health system where everyone is eligible, how do I remove/edit these such that they do not cause trouble?

Also, very silly question, if I just naively change the demographics for one of the states, can I simply use that to simulate data that follows Egypt's demographics? Or is there something under the hood that may override that? With my very untrained eyes, it looks like it could work, but also wanted to double check.

**Reply by jawalonoski** [reply]
For Universal Health, I'll follow up with some details. For your second question -- you're right, it will function without issue, the patient addresses will just look strange. And unless you add multiple providers, everyone will be visiting the same provider. As long as you don't mind the artificial-ness of that in your data, you'll be fine.

**Reply by nourabdelbaki** [reply]
The artificialness of it is not of a major concern for me now, so I think this will work out fine. Thank you so much for your help!
I will wait for the details on Universal health.

**Reply by jawalonoski** [reply, marked as answer]
For universal health...

1. Change `payers/insurance_companies.csv`

```
Id,Name,Address,City,State Headquarterd,Zip,Phone,States Covered,Ownership
10000,Universal,7500 Security Blvd,Baltimore,MD,21244,1-800-633-4227,*,Government
```

2. Change `payers/insurance_plans.csv`

```
Payer Id,Plan Id,Name,Services Covered,Deductible,Default Coinsurance,Default Copay,Monthly Premium,Max Out of Pocket,Medicare Supplement,Is ACA Plan,Income Based Premium,Eligibility Policy,Start Year,End Year,Priority Level,Notes
10000,10001,Universal Plan,*,0,1,0,0,0,FALSE,FALSE,FALSE,Generic,0,,1,
```

**Reply by nourabdelbaki** [reply]
Thank you so much!! Highly appreciate it!

**Comment by pmedvedev** [comment]
Hi @nourabdelbaki did you end up getting this to work? I've followed the instructions on https://github.com/synthetichealth/synthea-international but ran into the same issue. I've then tried changing the insurance files in the same way. That generated provider and organization data, but failed to make any patient info.
